### PR TITLE
[Week of Quality] Nightly build and testing of AzCopy on GOOS=Windows/MacOs/Ubuntu

### PR DESCRIPTION
**Pull request by mohsha-msft**
[Link to pipeline](https://dev.azure.com/azstorage/AzCopy-NextGen/_build?definitionId=30)

### CODE REVIEW

File: nightly-build.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    branches:
+      include:
+        - dev
+    always: true
+
+jobs:
+  - job: Nightly
+    # allow maximum build time, in case we have build congestion
+    timeoutInMinutes: 360
+    # Creating strategies for GOOS: Windows Server 2019 /macOS X Mojave 10.14/Ubuntu 20.04
```
Comment (zezha-msft) [inline nightly-build.yml:13]: can we use the latest version of MacOS instead? Big Sur?

File: nightly-build.yml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        displayName: 'Build MacOS'
+        condition: eq(variables.type, 'mac-os')
+
+      - task: DownloadSecureFile@1
+        name: ciGCSServiceAccountKey
+        displayName: 'Download GCS Service Account Key'
+        inputs:
+          secureFile: 'ci-gcs-dev.json'
+
+#      # Running Unit Tests on MacOS
```
Comment (zezha-msft) [inline nightly-build.yml:59]: remove commented parts

File: go.mod [hunk 3]
```diff
@@ -11,6 +11,7 @@ require (
 	github.com/danieljoos/wincred v1.0.1
 	github.com/go-ini/ini v1.41.0 // indirect
 	github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
+	github.com/google/uuid v1.1.1
```
Comment (zezha-msft) [inline go.mod:14]: where did this come from? is it used anywhere?

File: nightly-build.yml (hunk 1 above)
Comment (mohsha-msft) [reply nightly-build.yml:13]: Believe it or not but this is the latest which Azure supports :P

File: nightly-build.yml (hunk 2 above)
Comment (mohsha-msft) [reply nightly-build.yml:59]: I promise I'll do it in next week of quality. I may need to refer it so I haven't done it.

File: cmd/zt_test.go [hunk 4]
```diff
@@ -111,7 +111,7 @@ func generateName(prefix string, maxLen int) string {
 	var pcs [10]uintptr
 	n := runtime.Callers(1, pcs[:])
 	frames := runtime.CallersFrames(pcs[:n])
-	name := "TestFoo" // default stub "Foo" is used if anything goes wrong with this procedure
+	name := runtime.GOOS // default stub "Foo" is used if anything goes wrong with this procedure
```
Comment (adreed-msft) [inline cmd/zt_test.go:114]: Perhaps instead the stub should be TestFoo-GOOS?

Comment (adreed-msft) [reply cmd/zt_test.go:114]: Alternatively the attached comment should be changed.

File: nightly-build.yml (hunk 2 above)
Comment (adreed-msft) [reply nightly-build.yml:59]: Perhaps it'd be best to store them to the side somewhere safe like OneNote for right now. @zezha-msft?

**Review by adreed-msft — APPROVED** [review]

**Pull request merged**